### PR TITLE
feat: Add transparent appearance to ToolbarButton

### PR DESCRIPTION
**Pull request by tomi-msft**
This PR adds the value `transparent`  to the `appearance` prop of `ToolbarButton` and adds a story for it. There is a component in FAI that requires a `ToolbarButton` with a transparent appearanc, i.e, no background or border styling.

image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-toolbar-c990121c-d88f-4149-95f2-1666025abe74.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "feat: Add transparent to the appearance prop in ToolbarButton",
```
Comment (khmakoto) [inline change/@fluentui-react-toolbar-c990121c-d88f-4149-95f2-1666025abe74.json:3]: _nit:_

```suggestion
  "comment": "feat: Add transparent as an option for the appearance prop in ToolbarButton.",
```

File: packages/react-components/react-toolbar/stories/src/Toolbar/ToolbarTransparent.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { TextBold24Regular, TextItalic24Regular, TextUnderline24Regular } from '@fluentui/react-icons';
+import { Toolbar, ToolbarToggleButton } from '@fluentui/react-components';
+import type { ToolbarProps } from '@fluentui/react-components';
+
+export const transparent = (props: Partial<ToolbarProps>) => (
```
Comment (khmakoto) [inline packages/react-components/react-toolbar/stories/src/Toolbar/ToolbarTransparent.stories.tsx:6]:
```suggestion
export const Transparent = (props: Partial<ToolbarProps>) => (
```

**Review by khmakoto — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/Avatar-mount_base.html">632</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/Avatar-mount.html">663</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/Button-mount_base.html">312</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/Button-mount.html">308</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/Field-mount_base.html">1097</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/Field-mount.html">1121</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/FluentProvider-mount_base.html">728</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/FluentProvider-mount.html">712</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/FluentProviderWithTheme-mount_base.html">87</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/FluentProviderWithTheme-mount.html">90</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">34</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">34</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">89</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">77</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/MakeStyles-mount_base.html">858</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/MakeStyles-mount.html">865</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/Persona-mount_base.html">1750</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/Persona-mount.html">1682</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/SpinButton-mount_base.html">1396</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/SpinButton-mount.html">1397</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/SwatchPicker-mount_base.html">1629</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32205/perf-test-react-components/SwatchPicker-mount.html">1632</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-toolbar-c990121c-d88f-4149-95f2-1666025abe74.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-toolbar-c990121c-d88f-4149-95f2-1666025abe74.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32205&commitId=678d5ac374049b943cf3811d1d89233cc82370e8&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.normal.chromium.png|5| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: change/@fluentui-react-toolbar-c990121c-d88f-4149-95f2-1666025abe74.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+  "type": "patch",
```
Comment (ling1726) [inline change/@fluentui-react-toolbar-c990121c-d88f-4149-95f2-1666025abe74.json:2]: this change should be a minor - it's adding a new feature

File: packages/react-components/react-toolbar/library/src/components/ToolbarToggleButton/ToolbarToggleButton.types.ts [hunk 5]
```diff
@@ -6,7 +6,7 @@ import { ToggleButtonProps, ButtonSlots, ToggleButtonState } from '@fluentui/rea
  */
 export type ToolbarToggleButtonProps = ComponentProps<ButtonSlots> &
   Partial<Pick<ToggleButtonProps, 'disabled' | 'disabledFocusable' | 'size'>> & {
-    appearance?: 'primary' | 'subtle';
+    appearance?: 'primary' | 'subtle' | 'transparent';
```
Comment (ling1726) [inline packages/react-components/react-toolbar/library/src/components/ToolbarToggleButton/ToolbarToggleButton.types.ts:9]: please add VR tests for these new variants

Comment (tomi-msft) [reply packages/react-components/react-toolbar/library/src/components/ToolbarToggleButton/ToolbarToggleButton.types.ts:9]: VR tests have been added

**Review by layershifter — APPROVED** [review]

**Pull request merged**